include/ginkgo/core/matrix/identity.hpp

```diff
@@ -88,11 +88,12 @@ class Identity : public EnableLinOp<Identity<ValueType>>, public Transposable {
 template <typename ValueType = default_precision>
 class IdentityFactory
     : public EnablePolymorphicObject<IdentityFactory<ValueType>, LinOpFactory> {
-    friend class EnablePolymorphicObject<IdentityFactory, LinOpFactory>;
-
 public:
     using value_type = ValueType;
 
+    struct parameters_type
+        : enable_parameters_type<parameters_type, IdentityFactory> {};
+
```

Comment:
```suggestion
/**
     * Create the parameters from the property_tree.
     * Because this is directly tied to the specific type, the value/index type
     * settings within config are ignored and type_descriptor is only used
     * for children configs.
     *
     * @param config  the property tree for setting
     * @param context  the registry
     * @param td_for_child  the type descriptor for children configs. The
     *                      default uses the value type of this class.
     *
     * @return parameters
     */
    static parameters_type parse(const config::pnode& config,
                                 const config::registry& context,
                                 const config::type_descriptor& td_for_child =
                                     config::make_type_descriptor<ValueType>());
```
provide the same functionality when it support parse